hnn_core/gui/gui.py

```diff
@@ -1516,6 +1742,160 @@ def run_button_clicked(widget_simulation_name, log_out, drive_widgets,
                                           plot_type, {}, "plot")
 
 
+def _update_cell_params_vbox(cell_type_out, cell_parameters_list,
+                             cell_type, cell_layer):
+    cell_parameters_key = f"{cell_type}_{cell_layer}"
+    if "Biophysics" in cell_layer:
+        cell_parameters_key += f" {cell_type}"
+
+    if cell_parameters_key in cell_parameters_key:
+        cell_type_out.clear_output()
+        with cell_type_out:
+            display(cell_parameters_list[cell_parameters_key])
+
+
+def update_geometry_cell_params(net, cell_param_key, param_list):
+    cell_params = param_list
+    cell_type = f'{cell_param_key.split("_")[0]}_pyramidal'
+
+    sections = net.cell_types[cell_type].sections
+    # Soma
+    sections['soma']._L = cell_params[0].value
+    sections['soma']._diam = cell_params[1].value
+    sections['soma']._cm = cell_params[2].value
+    sections['soma']._Ra = cell_params[3].value
+
+    # Dendrite common parameters
+    dendrite_cm = cell_params[4].value
+    dendrite_Ra = cell_params[5].value
+
+    dendrite_sections = [
+        'apical_trunk', 'apical_1', 'apical_tuft', 'apical_oblique',
+        'basal_1', 'basal_2', 'basal_3'
+    ]
+
+    param_indices = [
+        (6, 7), (8, 9), (10, 11), (12, 13), (14, 15), (16, 17), (18, 19)]
+
+    # Dentrite
+    for section, indices in zip(dendrite_sections, param_indices):
+        sections[section]._L = cell_params[indices[0]].value
+        sections[section]._diam = cell_params[indices[1]].value
+        sections[section]._cm = dendrite_cm
+        sections[section]._Ra = dendrite_Ra
+
+
+def update_synapse_cell_params(net, cell_param_key, param_list):
```

Comment:
private function since it's not documented and may change in the future ...

Reply:
renamed as a private function